application-admintools-default/src/main/java/com/xwiki/admintools/internal/usage/SpamPagesProvider.java

```diff
@@ -102,20 +117,35 @@ public List<DocumentReference> getDocumentsOverGivenNumberOfComments(long maxCom
      * @throws QueryException if there are any exceptions while running the queries for data retrieval.
      */
     public List<DocumentReference> getCommentsForWiki(long maxComments, String searchedDocument, String wikiId)
-        throws QueryException
+        throws Exception
     {
         String searchString;
         if (searchedDocument == null || searchedDocument.isEmpty()) {
             searchString = "%";
         } else {
             searchString = String.format("%%%s%%", searchedDocument);
         }
-        return this.queryManager.createQuery("select obj.name from XWikiDocument as doc, BaseObject as obj "
-                + "where doc.fullName = obj.name and obj.className = 'XWiki.XWikiComments' "
-                + "and lower(doc.title) like lower(:searchString) "
-                + "group by obj.name having count(*) > :maxComments order by count(*) desc", Query.HQL).setWiki(wikiId)
-            .bindValue("maxComments", maxComments).bindValue("searchString", searchString)
-            .addFilter(currentLanguageFilter).addFilter(hiddenDocumentFilter).addFilter(documentFilter)
-            .addFilter(viewableFilter).execute();
+
+        SolrQuery solrQuery = new SolrQuery();
+
+        // Search for documents where the title contains the search string and have comments
+        solrQuery.setQuery("title:* AND object:XWiki.XWikiComments" + searchString); //
+        // Apply filters
+        solrQuery.addFilterQuery("hidden:false");
+        solrQuery.addFilterQuery("docviewable:true");
```

Comment:
We normally filter viewable pages after the query returns. I think you should use a `"secure"` query manager and enable `checkCurrentUser` as in [the script service](https://github.com/xwiki/xwiki-platform/blob/76b7341681e486a8155f308ea8583f0f349aeb22/xwiki-platform-core/xwiki-platform-query/xwiki-platform-query-manager/src/main/java/org/xwiki/query/script/QueryManagerScriptService.java#L117-L123).